dtls.c

```diff
@@ -542,25 +542,48 @@ static uint8 compression_methods[] = {
   TLS_COMPRESSION_NULL
 };
 
-/** returns true if the cipher matches TLS_ECDHE_ECDSA_WITH_AES_128_CCM_8 */
-static inline int is_tls_ecdhe_ecdsa_with_aes_128_ccm_8(dtls_cipher_t cipher)
+/**
+ * Get MAC length of cipher suite.
+ * \param cipher cipher suite
+ * \return MAC length of cipher. \c 0, if cipher is not supported.
+ */
+static inline int get_cipher_suite_mac_len(dtls_cipher_t cipher)
+{
+#ifdef DTLS_ECC
```

Comment:
For a little more future flexibility, I suggest restructuring this code to set a local variable which is then returned. This could go into a switch statement, e.g.  (not tested):

```c
int length;
switch (cipher) {
default:
  length = 0;
  break;
#ifdef DTLS_ECC
case TLS_ECDHE_ECDSA_WITH_AES_128_CCM_8:
  length = 8;
  break;
case TLS_ECDHE_ECDSA_WITH_AES_128_CCM:
  length = 16;
  break;
#endif /* DTLS_ECC */
#ifdef DTLS_PSK
case TLS_PSK_WITH_AES_128_CCM_8:
  length = 8;
  break;
case TLS_PSK_WITH_AES_128_CCM:
  length = 16;
  break;
#endif /* DTLS_PSK */
}
return length;
```

Or, even more flexible, with a table (not tested; `uint8_t` used to represent the size of the MAC tag to match the `tag_length` field of `dtls_ccm_params_t`):

```c
static inline uint8_t get_cipher_suite_mac_len(dtls_cipher_t cipher) {
  struct cipher_suite_param_t {
    uint16_t cipher_suite;
    uint8_t mac_length;
  };
  const struct cipher_suite_param_t cipher_suite_params[] = {
#ifdef DTLS_PSK
    { TLS_PSK_WITH_AES_128_CCM_8,          8 },
    { TLS_PSK_WITH_AES_128_CCM,           16 },
#endif /* DTLS_PSK */
#ifdef DTLS_ECC
    { TLS_ECDHE_ECDSA_WITH_AES_128_CCM_8,  8 },
    { TLS_ECDHE_ECDSA_WITH_AES_128_CCM,   16 },
#endif /* DTLS_ECC */
    /* TLS_NULL_WITH_NULL_NULL must always be the last entry as it
     * indicates the stop marker for the traversal of this table. */
    { TLS_NULL_WITH_NULL_NULL,             0 }
  };

  const struct cipher_suite_param_t *p = cipher_suite_params;
  while ((p->cipher_suite != cipher) &&
         (p->cipher_suite != TLS_NULL_WITH_NULL_NULL)) {
    p++;
  }
  return p->mac_length;
}
```

Reply:
According the 1. proposal, the switch over enum with internal `#ifdef` results in warnings. Therefore I assumed, that for this easy case, the simple `if` are a not to bad solution.
For future extensions you right,  that a parameter table isn't that bad.

Should I amend this PR?